lib/master.js

```diff
@@ -195,9 +195,35 @@ Master.prototype._startWorkers = function(remainingWorkers) {
                 type: 'config',
                 body: yaml.dump(self.config)
             });
+
+            var workerExit = function(code) {
```

Comment:
@d00rman This is equivalent to setting the cluster-level exit handler after all workers were started, so makes it possible to loose workers. 

Conceptually, there're 2 different exit reasons: 
- The worker failed to start during startup - this one should be handled by the `workerExit` handler
- The worker started up normally, but then died before all other workers were started. This one is handled by cluster-level exit handler. 

Unfortunately we can't separate one from another, so if we disable cluster-level handler during startup we will loose workers.